command/src/config.rs

```diff
@@ -214,13 +215,66 @@ pub struct FileAppConfig {
   pub certificate:       Option<String>,
   pub key:               Option<String>,
   pub certificate_chain: Option<String>,
-  pub backends:          Vec<String>,
+  pub backends:          Vec<BackendConfig>,
   pub sticky_session:    Option<bool>,
   pub https_redirect:    Option<bool>,
   #[serde(default)]
   pub send_proxy:        Option<bool>,
   #[serde(default)]
   pub expect_proxy:      Option<bool>,
+  #[serde(default)]
+  pub lb_policy: LoadBalancingAlgorithms,
+}
+
+#[derive(Debug,Copy,Clone,PartialEq,Eq,Hash, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum LoadBalancingAlgorithms {
+  RoundRobin,
+  Random,
+}
+
+impl Default for LoadBalancingAlgorithms {
+  fn default() -> Self {
+    LoadBalancingAlgorithms::RoundRobin
+  }
+}
+
+#[derive(Debug)]
+pub struct ParseErrorLoadBalancing;
+
+impl fmt::Display for ParseErrorLoadBalancing {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Cannot find the load balancing policy asked")
+    }
+}
+
+impl error::Error for ParseErrorLoadBalancing {
+    fn description(&self) -> &str {
+        "Cannot find the load balancing policy asked"
+    }
+
+    fn cause(&self) -> Option<&error::Error> {
+        None
+    }
+}
+
+impl FromStr for LoadBalancingAlgorithms {
```

Comment:
why do you need to implement `FromStr` when we already have serde's parsing?

Reply:
I need to implement `FromStr` for `structopt` in the `cli`. You can set the `load_balancing_policy` in the `ApplicationCmd`.

Reply:
Done